software/contrib/menu.py

```diff
@@ -12,39 +12,50 @@
-    "contrib.hello_world.HelloWorld",
-    "contrib.knob_playground.KnobPlayground",
-    "contrib.kompari.Kompari",
-    "contrib.logic.Logic",
-    "contrib.master_clock.MasterClock",
-    "contrib.noddy_holder.NoddyHolder",
-    "contrib.pams.PamsWorkout",
-    "contrib.piconacci.Piconacci",
-    "contrib.polyrhythmic_sequencer.PolyrhythmSeq",
-    "contrib.poly_square.PolySquare",
-    "contrib.probapoly.Probapoly",
-    "contrib.quantizer.QuantizerScript",
-    "contrib.radio_scanner.RadioScanner",
-    "contrib.scope.Scope",
-    "contrib.sequential_switch.SequentialSwitch",
-    "contrib.smooth_random_voltages.SmoothRandomVoltages",
-    "contrib.strange_attractor.StrangeAttractor",
-    "contrib.turing_machine.EuroPiTuringMachine",
-    "calibrate.Calibrate",
-]
-
+from collections import OrderedDict
+
+## Scripts that are included in the menu
+#
+#  Keys are the names displayed in the menu, values are the fully-qualified names
+#  of the classes to launch.  The classes MUST be EuriPiScript subclasses
+#
+#  The OLED can display up to 16 characters horizontally, so make sure the names fit
+#  that width requirement
+#
+# fmt: off
+EUROPI_SCRIPTS = OrderedDict([
+#   ["0123456789abcdef",  "contrib.spam.Eggs"],
+    ["Bernoulli Gates",   "contrib.bernoulli_gates.BernoulliGates"],
+    ["Coin Toss",         "contrib.coin_toss.CoinToss"],
+    ["Consequencer",      "contrib.consequencer.Consequencer"],
+    ["CVecorder",         "contrib.cvecorder.CVecorder"],
+    ["Diagnostic",        "contrib.diagnostic.Diagnostic"],
+    ["EnvelopeGen",       "contrib.envelope_generator.EnvelopeGenerator"],
+    ["Euclid",            "contrib.euclid.EuclideanRhythms"],
+    ["Hamlet",            "contrib.hamlet.Hamlet"],
+    ["HarmonicLFOs",      "contrib.harmonic_lfos.HarmonicLFOs"],
+    ["HelloWorld",        "contrib.hello_world.HelloWorld"],
+    ["KnobPlayground",    "contrib.knob_playground.KnobPlayground"],
+    ["Kompari",           "contrib.kompari.Kompari"],
+    ["Logic",             "contrib.logic.Logic"],
+    ["MasterClock",       "contrib.master_clock.MasterClock"],
+    ["NoddyHolder",       "contrib.noddy_holder.NoddyHolder"],
+    ["Pam's Workout",     "contrib.pams.PamsWorkout"],
+    ["Piconacci",         "contrib.piconacci.Piconacci"],
+    ["PolyrhythmSeq",     "contrib.polyrhythmic_sequencer.PolyrhythmSeq"],
+    ["PolySquare",        "contrib.poly_square.PolySquare"],
+    ["Probapoly",         "contrib.probapoly.Probapoly"],
+    ["Quantizer",         "contrib.quantizer.QuantizerScript"],
+    ["RadioScanner",      "contrib.radio_scanner.RadioScanner"],
+    ["Scope",             "contrib.scope.Scope"],
+    ["Seq. Switch",       "contrib.sequential_switch.SequentialSwitch"],
+    ["Smooth Rnd Volts",  "contrib.smooth_random_voltages.SmoothRandomVoltages"],
+    ["StrangeAttractor",  "contrib.strange_attractor.StrangeAttractor"],
+    ["Turing Machine",    "contrib.turing_machine.EuroPiTuringMachine"],
```

Comment:
[discussion] This change effectively removes the responsibility for setting the display_name from the individual scripts to this file. This means that we should probably remove it from `europi_script.py` and all of the individual scripts (and update any docs that reference it). I had envisioned that as a part of this PR, do you think that it should be a separate effort?

Reply:
There _may_ be other uses for the `display_name` in the future. I don't necessarily think they need to be gotten rid of, though a lot of the `contrib` scripts seem to omit it anyway.

My vote would be to leave the `display_name`s alone for now, and removing them from scripts that have them (or adding them to the scripts that omit them) can probably be a separate effort.